tests/src/main/java/org/hibernate/beanvalidation/tck/tests/validation/groupconversion/RegisteredAddresses.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Bean Validation TCK
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.beanvalidation.tck.tests.validation.groupconversion;
+
+import java.time.Year;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.validation.Valid;
+import javax.validation.groups.ConvertGroup;
+import javax.validation.groups.Default;
+
+/**
+ * @author Guillaume Smet
+ */
+public class RegisteredAddresses {
+
+	private final Map<Year, List<@Valid @ConvertGroup(from = Default.class, to = BasicPostal.class) Address>> mainAddresses = new HashMap<>();
+
+	private final Map<Year, List<Address>> shipmentAddresses = new HashMap<>();
+
+	private final Map<Year, List<@Valid @ConvertGroup.List({
+			@ConvertGroup(from = Default.class, to = BasicPostal.class),
+			@ConvertGroup(from = Complex.class, to = ComplexPostal.class)
+	}) Address>> preferredShipmentAddresses = new HashMap<>();
+
+	private final Map<Year, List<@Valid @ConvertGroup.List({
+			@ConvertGroup(from = Default.class, to = BasicPostal.class),
+			@ConvertGroup(from = BasicPostal.class, to = ComplexPostal.class)
+	}) Address>> officeAddresses = new HashMap<>();
+
+	private final Map<Year, List<@Valid @ConvertGroup(from = Default.class, to = PostalSequence.class) Address>> weekendAddresses = new HashMap<>();
+
+	public RegisteredAddresses() {
+	}
+
+	@Valid
+	@ConvertGroup(from = Default.class, to = BasicPostal.class)
+	public RegisteredAddresses(Map<Year, List<@Valid @ConvertGroup(from = Default.class, to = BasicPostal.class) Address>> mainAddresses) {
```

Comment:
Note that we should probably add a test for the case you have in mind but to be honest I'm still not sure how it should behave and we should probably discuss it next week when we have more time.